spec/requests/api/v1/digital_objects_spec.rb

```diff
@@ -118,18 +120,71 @@
     end
 
     context "logged in" do
+      let(:publication_adapter) { instance_double(Hyacinth::Adapters::PublicationAdapter::Abstract) }
+      let(:configured_publication_adapter) { Hyacinth.config.publication_adapter }
       before do
+        configured_publication_adapter
+        Hyacinth.config.publication_adapter = publication_adapter
         sign_in_project_contributor to: [:delete_objects, :read_objects], project: authorized_project
-        delete "/api/v1/digital_objects/#{authorized_object.uid}"
       end
 
-      xit 'returns 200' do
+      after do
+        Hyacinth.config.publication_adapter = configured_publication_adapter
+      end
+
+      it 'returns 204' do
+        expect(publication_adapter).to receive(:unpublish).with(authorized_publish_target, digital_object_matcher).and_return([true, []])
+        delete "/api/v1/digital_objects/#{authorized_object.uid}"
         expect(response.status).to be 204
         get "/api/v1/digital_objects/#{authorized_object.uid}"
         expect(response.status).to be 404
       end
-      xit "return a single Digital Object with the expected fields" do
-        expect(JSON.parse(response.body)).to be_blank
+      it 'unpublishes the object on delete' do
+        expect(publication_adapter).to receive(:unpublish).with(authorized_publish_target, digital_object_matcher).and_return([true, []])
+        delete "/api/v1/digital_objects/#{authorized_object.uid}"
+      end
+      it "return blank response entity" do
+        expect(publication_adapter).to receive(:unpublish).with(authorized_publish_target, digital_object_matcher).and_return([true, []])
+        delete "/api/v1/digital_objects/#{authorized_object.uid}"
+        expect(response.body).to be_blank
+      end
+    end
+  end
+
+  describe 'POST /api/v1/digital_objects/:id/publish' do
+    include_examples 'requires user to have correct permissions' do
+      let(:request) do
+        post "/api/v1/digital_objects/#{authorized_object.uid}/publish"
+      end
+    end
+
+    context "logged in" do
+      let(:publication_adapter) { instance_double(Hyacinth::Adapters::PublicationAdapter::Abstract) }
+      let(:configured_publication_adapter) { Hyacinth.config.publication_adapter }
```

Comment:
Can use `let!` here to automatically run this.

Reply:
that's awesome & new to me